src/staging/src/components/park/parkAccessStatus.js

```diff
@@ -20,18 +21,65 @@ const ICONS = {
   "yellow": yellowStatusIcon,
   "red": redStatusIcon,
 }
+
 export default function ParkAccessStatus({ advisories }) {
+
+  const data = useStaticQuery(
+    graphql`
+      {
+        allStrapiAccessStatuses {
+          edges {
+            node {
+              id
+              color
+              accessStatus
+              precedence
+            }
+          }
+        }
+      }
+    `
+  )
+
+  const accessStatusList = data?.allStrapiAccessStatuses.edges;
+
   const classes = useStyles()
   let parkStatusIcon = blueStatusIcon
   let parkStatusText = "Open to public access"
 
-  const accessStatuses = advisories.filter(advisory => advisory.accessStatus).map(advisory => {
-    return {
-      precedence: advisory.accessStatus.precedence,
-      color: advisory.accessStatus.color,
-      text: advisory.accessStatus.accessStatus,
+  // unfortunately, incoming advisories from parks details and explore pages are structured differently.
+  // we need to differentiate between the two structures. 
+
+  let accessStatuses = [];
+
+  for (let advisory of advisories) {
+    if (advisory.accessStatus) {
+      if (advisory.accessStatus.precedence) {
+        // advisory is coming from parks details page
+        accessStatuses.push({
+          precedence: advisory.accessStatus.precedence,
+          color: advisory.accessStatus.color,
+          text: advisory.accessStatus.accessStatus,
+        });
+      } else {
+        // advisory is coming from explore page
+        // get accessStatus based on precedence
+        let thisStatus = accessStatusList.filter(status => {
+          return status.node.precedence === advisory.accessStatus;
```

Comment:
I think this match up should be done by id as that's what the value of `advisory.accessStatus` is (although `id` and `precedence` happen to be the same in most cases). Gatsby calls the id from Strapi `strapiId` so probably `status.node.strapiId === advisory.accessStatus` (you'll have to add `strapiId` to the query as well). The value of `id` includes some text that Gatsby adds so I don't think that will match.

Also up to you but I think `accessStatusList.find` is a minor improvement on filter here.

Reply:
Agree with all this (I wasnt even aware of the `accessStatus` vs `strapiId` business). Updated in `dbcc220`